Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -1961,20 +1961,32 @@ static MTLPixelFormat pixelFormat(WGPUTextureFormat textureFormat)
     }
 }
 
-static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures)
+static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures, bool isBackedByIOSurface)
 {
+    // IOSurface-backed texture must be MTLStorageModeManaged, else the driver will assert with
+    // "Texture Descriptor Validation IOSurface textures must use MTLStorageModeManaged"
+    if (isBackedByIOSurface)
+#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
+        return MTLStorageModeManaged;
+#else
+        // MTLStorageModeManaged is not supported on iOS.
+        return MTLStorageModeShared;
+#endif
```

Comment:
We have a block down below that has this code duplicated. Can we deduplicate it somehow?

I'm also not so sure about the usefulness of the comment.

Reply:
I like to document stuff that I don't know (because I only know that when compiling for the iOS simulator). But I can remove it.

Reply:
I tried looking into deduplicating the code but I found that it makes the logic hard to read. I think the current way of doing is straightforward - if it's backed by an IOSurface then we just make it Managed/Shared.